### ISSUE TITLE
[BUG] Inconsistent items are generated when an item is created and when an item is fetched from the database

### ISSUE THREAD

**Issue by ptejada** [dynamoose/dynamoose#1479]
### Summary:
When creating a new item using a model the field aliases are not accessible. When fetching an item the main field itself is not accessible only its alias is. 

**Disclaimer:** I have only tried this in dynamodb local and with dynamoose v3 so can't say if this is the expected behavior or is bug. It was suggested in Slack that ticket is created https://dynamoose.slack.com/archives/CG4B7RL8N/p1661138601979229

### Code sample:
```js
    const User = dynamoose.model("User", {"pk": {"type": String, "alias": "email"}});
    new dynamoose.Table("User", [User]);

    const user1 = await User.create({ "email": "joe@doe.com" })
    const user2 = await User.create({ "pk": "jane@doe.com" })

    console.log(user1, user1.email) // User { pk: 'joe@doe.com' } undefined
    console.log(user2, user2.email) // User { pk: 'jane@doe.com' } undefined

    const user3 = await User.get('jane@doe.com')
    console.log(user3, user3.pk) // User { email: 'jane@doe.com' } undefined
```

### Current output and behavior (including stack trace):
The current output is inlined in the code example above

### Expected output and behavior:
The expected behavior is for `Model.create()` and `Model.get()` generate consistent items. Both the DDB field and the alias should be accessible in the generated items.

### Environment:

Operating System: Ubuntu
Operating System Version:  20.04.2 LTS
Node.js version (`node -v`): v14.19.0
NPM version: (`npm -v`): 8.6.0
Dynamoose version: 3.0.0

### Other:
- [x] I have read through the Dynamoose documentation before posting this issue
- [x] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [ ] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [x] I have tested the code provided and am confident it doesn't work as intended
- [x] I have filled out all fields above
- [x] I am running the latest version of Dynamoose

**Comment by fishcharlie** [comment]
> The expected behavior is for Model.create() and Model.get() generate consistent items

I tend to agree on this. However, slightly concerned this is a breaking change... @andrewda what are your thoughts here?

> Both the DDB field and the alias should be accessible in the generated items.

Incorrect. This is not the expected behavior. According to the [Dynamoose documentation](https://dynamoosejs.com/guide/Schema#map-string--string): `When retrieving data from DynamoDB, the attribute will be renamed to this property name, or the first element of the array if it is an array`. Specifically, notice how it says `renamed`, not `added` or `appended`, but `renamed`.

There are many reasons for this design choice. For one, having duplicated attributes in an object would be **extremely** confusing. Imagine logging out the object and seeing the following:

```json
{
    "pk": "joe@doe.com",
    "email": "joe@doe.com"
}
```

That simply doesn't make sense at all. Those are 1 property, and should be treated, and displayed as such.

**Comment by ptejada** [comment]
Hey @fishcharlie 

Thank you for the feedback. Regardless, of the expected behavior, it just needs to be consistent. Especially when using the TS typing system. For example, notice how I have to add `pk` to the UserType here

```js
    class UserType extends Item {
      pk: String
      email: String
    }

    const User = dynamoose.model<UserType>("User", {"pk": {"type": String, "alias": ["email"]}});
```

`User.create()` and `User.get()` will return an object that is expected to be of the `UserType` shape however only one of `pk` or `email` will be accessible at any given time and not both.

Since you are only making the alias accessible and not the main underlying field as well, I can see how returning an item from `User.create()` with the alias resolved is a breaking change, because you would have to remove access to the field that is accessible today.

In a world where a field alias is prioritized in the `Model` and `Item` interface ideally `pk` is not allowed to be used anywhere, not even when creating or fetching a record. Is just a point of confusion. Even supporting multiple aliases when only one of them is made accessible from the item instance can be confusing.

In a world where both the field and aliases are accessible from the item instance, everyone can choose their own poison. From the serialization perspective, it should all stay the same, the underlying field will still be used when generating the DDB items and only the default alias will still be included in JSON or as enumerable attributes.

**Comment by fishcharlie** [comment]
@ptejada That's why I said:

> I tend to agree on this.

**Comment by halan** [comment]
https://github.com/dynamoose/dynamoose/issues/1378 is related with this. Currently the return of create/save is how the data was sent to DynamoDB, not the Schema processed fields.

I'm investigating that behavior and planning to write a PR to v4 doing a similar process that we do on 'ItemRetriever'

**Comment by wjensen-godaddy** [comment]
> The expected behavior is for Model.create() and Model.get() generate consistent items

+1

(And thanks for all your hard work on this project!)

**Comment by ptejada** [comment]
Is there any trick we can use today to transform the items?

**Comment by hpinheiro-godaddy** [comment]
There is an undocumented/unrecommended way.
```ts
user1.conformToSchema({
  customTypesDynamo: true,
  checkExpiredItem: true,
  saveUnknown: true,
  modifiers: ["get"],
  type: "fromDynamo"
});
```

This is what is used on [`get`](https://github.com/dynamoose/dynamoose/blob/main/packages/dynamoose/lib/Model/index.ts#L426). I'm not sure and haven't tested your specific case, but that function supports [some settings to enable/disable features on schema](https://github.com/dynamoose/dynamoose/blob/main/packages/dynamoose/lib/Item.ts#L600-L617) transformation.

**Comment by ptejada** [comment]
@hpinheiro-godaddy 

Thank you for the tip. Unfortunately, it doesn't do anything. Ex:
```js
    // Model + Schema
    const Model = dynamoose.model('test', {
      pk: {
        type: String,
        default: makeId,
        map: 'id'
      },
      sk: {
        type: String,
        default: makeId,
        map: 'range'
      }
    })

    // New Item
    const item = await Model.create({})

    // Attempt to transformation
    console.log(await item.conformToSchema({
      customTypesDynamo: true,
      checkExpiredItem: true,
      saveUnknown: true,
      modifiers: ["get"],
      type: "fromDynamo"
    }))

   // test { pk: '915aftZW74RRAG6', sk: 'Nyp06V6ZzY6Hq7V' }


```

**Comment by hpinheiro-godaddy** [comment]
try to include `mapAttributes: true`

**Comment by ptejada** [comment]
Beautiful! Thank you @hpinheiro-godaddy! Using `mapAttributes: true` works 👍.

I removed some of the other settings that I don't know what they do ended up with this:

```js
console.log(await item.conformToSchema({     
      mapAttributes: true,      
      modifiers: ["get"],
      type: "fromDynamo"
    }))

   //  test { id: '4yt0iHubQJoMJiy', range: '6P4u3rVVTlQBiFr' }
```

**Comment by Hideman85** [comment]
Having the same issue. when create is called the set functions are called but not the get one after saved.
Result is it is not in the right format.
Sample:
```js
const schema = {
  id: {
    type: String,
    hashKey: true,
  },
  set: {
    type: Set,
    schema: [String],
    //  Imagine in our code we want to always manipulate arrays but save a unique set in DB
    get: val => Array.from(val).sort(),
    set: val => Array.isArray(val) ? new Set(val) : val,
  },
}
```

**Comment by hpinheiro-godaddy** [comment]
try `modifiers: ["get"]`
